### PR TITLE
@types/node - NodeJS.BufferEncoding

### PR DESCRIPTION
**Pull request by dsherret**
This change ~~exports `BufferEncoding` and `BufferConstructor` from `node:buffer`~~ adds a `NodeJS.BufferEncoding` type which is the same as the `BufferEncoding` global. The reason is because in Deno these Node.js global types (except `NodeJS`) are not accessible globally in Deno typescript code, so they need to be imported.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/node/buffer.d.ts [hunk 1]
```diff
@@ -113,7 +113,7 @@ declare module 'buffer' {
      * @param id A `'blob:nodedata:...` URL string returned by a prior call to `URL.createObjectURL()`.
      */
     export function resolveObjectURL(id: string): Blob | undefined;
-    export { Buffer };
+    export { Buffer, BufferEncoding, BufferConstructor };
```
Comment (dsherret) [inline types/node/buffer.d.ts:116]: FWIW, I'm more interested in making `BufferEncoding` accessible. I evaluated all the types within `@types/node` and it seems these are the only two global types that aren't accessible in Deno via an export from a `node:x` module.

**Comment by Semigradsky** [comment]
I am not really understand these changes.
- `BufferEncoding` already available, it is in global scope.
- `BufferConstructor` already available via `typeof Buffer`.
image

**Comment by Semigradsky** [comment]
> The reason is because in Deno these Node.js global types are not accessible globally in Deno typescript code, so they need to be imported.

Why it's not accessible in Deno typescript code? Could you provide link to issue, example of code or smth?

**Comment by dsherret** [comment]
@Semigradsky in Deno there are two global scopes. A Deno global scope and an additive global scope for Node. This allows us to do stuff like have a web standard `setTimeout` and not make Node specific globals like `process` and `Buffer` global. Instead, people need to import those globals from the `node:` specific modules:

image

In the case of the `BufferEncoding` type, it is useful to be able to use it conveniently so that was the main reason for my PR (so it works similarly now to `Buffer` where even though it's a global type, it's also accessible on the import). With the case of `BufferConstructor`, it is less useful, but I added it for consistency because it's also a global in `@types/node` for some reason (so not accessible in Deno TypeScript code). That said, I think people just doing `typeof Buffer` is better as you mentioned so I have removed it from this PR.

**Comment by dsherret** [comment]
Actually, looking at the code more it looks like this should go on NodeJS instead (that's accessible in Deno). I will update this PR.

**Review by Semigradsky — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]
@dsherret thanks!

**Comment by Semigradsky** [comment]
Ready to merge

**Pull request merged**